vsifile/io/s3.py

```diff
@@ -14,14 +13,9 @@
 
 
 def _find_bucket_region(bucket: str) -> Optional[str]:
-    try:
-        logger.debug("VSIFILE_INFO_HEADER_OUT: GET")
-        response = urllib.request.urlopen(f"https://{bucket}.s3.amazonaws.com")
-        return response.getheader("x-amz-bucket-region")
-    except urllib.error.HTTPError:
-        pass
-
-    return None
+    logger.debug("VSIFILE_INFO_HEADER_OUT: GET")
+    response = httpx.get(f"https://{bucket}.s3.amazonaws.com")
```

Comment:
urllib will raise an error on 403 response (e.g for private bucket) and thus not return headers

by using `httpx` we can get it even it the response is not 200 😄